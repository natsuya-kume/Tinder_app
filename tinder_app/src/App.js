import React, { useEffect, useState } from "react";
import Card from "./Components/Card";
import "../src/App.css";

const App = () => {
  // 一番上の階層でuserDataを取得
  const [userData, setUserData] = useState([]);
  useEffect(() => {
    const url =
      "https://my-json-server.typicode.com/natsuya-kume/user_data/users";
    fetch(url)
      .then((res) => res.json())
      .then((data) => setUserData(data));
  }, []);

  return (
    <div className="app">
      <Card userData={userData} />
    </div>
  );
};
export default App;
