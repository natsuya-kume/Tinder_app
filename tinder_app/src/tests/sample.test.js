import React from "react";
import renderer from "react-test-renderer";
import App from "../App";

it("UI test", () => {
  const component = renderer.create(<App />);
  expect(component).toMatchSnapshot();
});
