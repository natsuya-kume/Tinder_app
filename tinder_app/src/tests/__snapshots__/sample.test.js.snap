// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI test 1`] = `
<div
  className="app"
>
  <div>
    <link
      href="https://fonts.googleapis.com/css?family=Damion&display=swap"
      rel="stylesheet"
    />
    <link
      href="https://fonts.googleapis.com/css?family=Alatsi&display=swap"
      rel="stylesheet"
    />
    <h1>
      Tinder_app
    </h1>
    <div
      className="cardContainer"
    >
      <div
        className="empty"
      >
        <h3>
          empty!
        </h3>
      </div>
    </div>
    <div
      className="buttons"
    >
      <button
        onClick={[Function]}
      >
        スキップ
      </button>
      <button
        onClick={[Function]}
      >
        いいね!
      </button>
    </div>
    <h2
      className="infoText"
    >
      Swipe the Card or Click Button!
    </h2>
  </div>
</div>
`;
